File: C#高级编程(第7版)源代码/502259 Code CH53/FSharp/ExceptionDemo.fs
namespace Wrox.ProCSharp.Languages
open System
open System.IO

type ExceptionTest() as this =
    member this.Method(o : obj) =
        if o = null then
            raise(ArgumentException("error"))
    member this.Foo = 
        try
            try
                this.Method(null)
            with
                | :? ArgumentException as ex -> printfn "%s" ex.Message
                | :? IOException as ex -> printfn "%s" ex.Message
        finally
            printfn "finally"

//
//public void Method(Object o)
//{
//   if (o == null)
//      throw new ArgumentException("Error");
//}
//public void Foo()
//{
//   try
//   {
//      Method(null);
//   }
//   catch (ArgumentException ex)
//   { }
//   catch (Exception ex)
//   { }
//   finally
//   { }
//}

File: C#高级编程(第7版)源代码/502259 Code CH53/CSharp/ExceptionDemo.cs
using System;
using System.Collections.Generic;
using System.Text;
using Message = System.Messaging.Message;

namespace CSharp
{

    public class ExceptionDemo
    {
        public void ThrowMethod()
        {
            throw new ArgumentException("error");
        }

        public void Handler()
        {
            try
            {
                ThrowMethod();
            }
            catch (ArgumentException ex)
            {
                Console.WriteLine(ex.Message);
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                Console.WriteLine("Finally");
            }
        }
    }
}
